platform/plugins/workspaces/index.ts

```diff
@@ -0,0 +1,59 @@
+import { Observable } from 'rxjs';
+
+import { KibanaPlugin } from '../../server/plugins/types';
+import { KibanaPluginFeatures } from '../../types';
+import { Logger } from '../../logger'
+import { Root } from '../../root'
+import { WorkspacesConfig } from './WorkspacesConfig';
+
+export const configPath = ['xpack', 'workspaces'];
+
+export const dependencies = [];
+
+export const plugin = class implements KibanaPlugin<void> {
+  log: Logger;
+  kibanaInstance?: Root;
+  config$: Observable<WorkspacesConfig>;
+
+  constructor(readonly kibana: KibanaPluginFeatures) {
+    this.log = kibana.logger.get();
+
+    this.config$ = kibana.config.create(WorkspacesConfig);
+  }
+
+  async start() {
+    const workspacesConfig = await this.config$
+      .first()
+      .toPromise();
+
+    if (!workspacesConfig.enabled) {
```

Comment:
(the pluginsystem doesn't handle `some.config.path.enabled: false` yet, but I've started on working that in locally)